packages/vscode-extension/src/treeview/environmentTreeItem.ts

```diff
@@ -69,13 +69,14 @@ export class EnvironmentNode extends DynamicNode {
 
   public async getTreeItem(): Promise<vscode.TreeItem> {
     const envInfo = await this.getCurrentEnvInfo(this.identifier);
-    const isSpfxProject = ext.workspaceUri ? isSPFxProject(ext.workspaceUri.fsPath) : false;
-
     this.iconPath = envInfo === EnvInfo.ProvisionedRemoteEnv ? provisionedIcon : nonProvisionedIcon;
+
+    const children = await this.getChildren();
```

Comment:
I think we'd better not calling `getChildren()` directly because VSCode will call this function if collapsibleState is Expanded. So the function would be called twice. The time cost is relatively small here but it's better to cache the children data.

Reply:
Thanks! It makes sense, let me take a further look at it.

Reply:
Updated, please take a look.